xarray/testing/strategies.py

```diff
@@ -62,6 +64,15 @@ def supported_dtypes() -> st.SearchStrategy[np.dtype]:
     )
 
 
+# TODO: add datetime64[ns]
+def pandas_index_dtypes() -> st.SearchStrategy[np.dtype]:
+    return (
+        npst.integer_dtypes(endianness="<", sizes=(32, 64))
+        | npst.unsigned_integer_dtypes(endianness="<", sizes=(32, 64))
+        | npst.floating_dtypes(endianness="<", sizes=(32, 64))
```

Comment:
Really little-endian, or 'machine-native byteorder happens to be' little-endian?  A comment to distinguish would be nice.